pydicom/tests/test_valuerep.py

```diff
@@ -889,11 +891,31 @@ def test_valid_value(self, disable_value_validation):
         assert 42 == IS("42.0")
         assert 42 == IS(42.0)
 
+    def test_float_value(self):
+        """Read binary value of IS that is actually a float"""
+        # from issue #1661
+        # Create BytesIO with single data element for Exposure (0018,1152)
+        #   length 4, value "14.5"
+        bin_elem = b"\x18\x00\x52\x11\x04\x00\x00\x0014.5"
+        with BytesIO(bin_elem) as bio:
+            ds = read_dataset(bio, True, True)
+        assert isinstance(ds.Exposure, ISfloat)
+        assert ds.Exposure == 14.5
+
+        # Strict checking raises an error
+        with pytest.raises(ValueError):
+            _ = IS("14.5", validation_mode=config.RAISE)
+        with pytest.raises(TypeError):
+            _ = IS(14.5, validation_mode=config.RAISE)
+
```

Comment:
This seems a bit odd to me to have two different Errors for the same value, but that is the way it is done currently.  Perhaps to be revisited for the grand redesign of validation settings?

Reply:
I remember that I was not happy about this either, though forgot why I didn't change it...

Reply:
In part I guess there are two branches, so to speak, because some are by string processing (regex) and others are by value. Another thing to think about in harmonizing validation.